docs/installation/webserver/nginx.md

```diff
@@ -6,106 +6,263 @@ Nginx: Configuring the virtual host
+path locations relevant to your system.
+
+
+### Common
+
+Common, or "global", configuration settings are stored in a directory under
+`/etc/nginx/`. For our examples we use the `/etc/nginx/global/` directory for
+no other reason besides semantics.
+
+This section contains details on the following files:
+
+| File name             | Description |
+| --------------------- | ----------- |
+| `bolt.conf`           | Bolt specific routes
+| `restrictions.conf`   | Files & directories to block access to
+| `php-fpm.conf`        | PHP-FPM configuration
+| `acme-challenge.conf` | (optional) Let's Encrypt Certbot challenges
+
+
+#### `bolt.conf`
 
+The `bolt.conf` file define location maches common to all of your Bolt sites on
+a host.
+
+```nginx
+# Redirect requests for */index.php to the same route minus the "index.php"
+if ($request_uri ~* "^(.*/)index\.php(?>/|)(.*)$") {
+    return                     301 $1$2;
+}
+
+# Default prefix match, as all URIs begin with /
+location / {
+    try_files                  $uri $uri/ /index.php?$query_string;
+}
+
+# Bolt backend access
+#
+# NOTE: If you set a custom branding path, you will need to change '/bolt/'
+#       here to match
+location ~* /bolt/(.*)$ {
+    try_files                  $uri $uri/ /index.php?$query_string;
+}
+
+# Generated thumbnail images
+location ~* /thumbs/(.*)$ {
+    access_log                 off;
+    log_not_found              off;
+    expires                    max;
+    add_header                 Pragma public;
+    add_header                 Cache-Control "public, mustrevalidate, proxy-revalidate";
+
+    try_files                  $uri $uri/ /index.php?$query_string;
+}
+
+# Don't log, and do cache, asset files
+location ~* ^.+\.(?:atom|bmp|bz2|css|doc|eot|exe|gif|gz|ico|jpe?g|jpeg|jpg|js|map|mid|midi|mp4|ogg|ogv|otf|png|ppt|rar|rtf|svg|svgz|tar|tgz|ttf|wav|woff|woff2|xls|zip)$ {
+    access_log                 off;
+    log_not_found              off;
+    expires                    max;
+    add_header                 Pragma public;
+    add_header                 Cache-Control "public, mustrevalidate, proxy-revalidate";
+}
+
+# Don't create logs for favicon.ico, robots.txt, or rss.jpg requests
+location = /(?:favicon.ico|robots.txt|rss.jpg) {
+    log_not_found              off;
+    access_log                 off;
+}
 ```
-fastcgi_pass unix:/path/to/php-fpm/socket.sock;
+
+
+#### `restrictions.conf`
+
+The `restrictions.conf` file defines a common set of restrictions for a Bolt
+site.
+
+```nginx
+# Block access to "hidden" files, i.e. file names that begin with a dot
+location ~ /\. {
+    deny                      all;
+}
+
+# Block PHP files from being run in upload (files), app, theme and extension directories
+location ~* /(?:extensions|files|theme)/(.*)\.php$ {
+    deny                      all;
+}
+
+# Block access to Sqlite database files, Apache .htaccess & .htpasswd files
+location ~ /\.(?:db|htaccess|htpasswd)$ {
```

Comment:
Yes, see point about "helpful" addition … if it irks too much I can just drop it